src/Controls/src/Core/Platform/Android/Extensions/RecyclerViewExtensions.cs

```diff
@@ -11,8 +11,8 @@ public static class RecyclerViewExtensions
 		public static void UpdateSelection(this RecyclerView recyclerView, SelectableItemsView selectableItemsView)
 		{
 			var mode = selectableItemsView.SelectionMode;
-
-			var adapter = (recyclerView.GetAdapter() as SelectableItemsViewAdapter<SelectableItemsView, IItemsViewSource>);
+			//TODO: on NET7 implement a ISelectableItemsViewAdapter interface on the adapter
+			var adapter = recyclerView.GetAdapter() as ReorderableItemsViewAdapter<ReorderableItemsView, IGroupableItemsViewSource>;
```

Comment:
this cast was failing, and that's why clearing selection wasn't working. We need to find a better way to fix this cast, for example implement a Interface on the SelectableItemsViewAdapter